packages/edit-site/src/components/global-styles/font-size.js

```diff
@@ -0,0 +1,257 @@
+
+	const hasUnits = typeof fontSize.size === 'string';
+
+	// Whether fluid is true or an object, set it to true, otherwise false.
+	const isFluid = !! fontSize.fluid ?? false;
+
+	// Whether custom fluid values are used.
+	const isCustomFluid = typeof fontSize.fluid === 'object';
+
+	// Initialize the ref with the prop value on the first render
+	const initialFontSizeRef = useRef( fontSize.size );
+
+	// Memoized initial value of fontSize
+	const initialFontSize = initialFontSizeRef.current;
+
+	const handleNameChange = ( value ) => {
+		updateFontSize( 'name', value );
+	};
+
+	const handleFontSizeChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? initialFontSize;
+		updateFontSize( 'size', newValue );
+	};
+
+	const handleFluidChange = ( value ) => {
+		const newValue = value === 'false' ? false : true;
+		updateFontSize( 'fluid', newValue );
+	};
+
+	const handleCustomFluidValues = ( value ) => {
+		if ( value ) {
+			// If custom values are use init the values with the current ones.
+			updateFontSize( 'fluid', {
+				min: fontSize.size,
+				max: fontSize.size,
+			} );
+		} else {
+			// If custom fluid values are disabled, set fluid to true.
+			updateFontSize( 'fluid', true );
+		}
+	};
+
+	const handleMinChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
+		updateFontSize( 'fluid', { ...fontSize.fluid, min: newValue } );
+	};
+
+	const handleMaxChange = ( value ) => {
+		// If the user is resetting the value, use the initial value.
+		const newValue = value ?? fontSize.size;
+		updateFontSize( 'fluid', { ...fontSize.fluid, max: newValue } );
+	};
+
+	const updateFontSize = ( key, value ) => {
+		const newFontSizes = sizes.map( ( size ) => {
+			if ( size.slug === slug ) {
+				return { ...size, [ key ]: value }; // Create a new object with updated key
+			}
+			return size;
+		} );
+
+		setFontSizes( {
+			...fontSizes,
+			theme: newFontSizes,
+		} );
+	};
+
+	const handleRemoveFontSize = () => {
+		// Navigate to the font sizes list.
+		goBack();
+
+		const newFontSizes = sizes.filter( ( size ) => size.slug !== slug );
+		setFontSizes( {
+			...fontSizes,
+			theme: newFontSizes,
+		} );
+	};
+
+	return (
+		<>
+			<HStack justify="space-between" align="flex-start">
+				<ScreenHeader
+					title={ fontSize.name }
+					description={ sprintf(
+						/* translators: %s: font size preset name. */
+						__( 'Manage the font size %s.' ),
+						fontSize.name
+					) }
+				/>
+				<FlexItem>
+					<Spacer marginTop={ 3 } marginBottom={ 0 } paddingX={ 4 }>
+						<DropdownMenu
+							trigger={
+								<Button
+									size="small"
+									icon={ moreVertical }
+									label={ __( 'Menu' ) }
+								/>
+							}
+						>
+							<DropdownMenuItem onClick={ handleRemoveFontSize }>
+								<DropdownMenuItemLabel>
+									{ __( 'Remove font size' ) }
+								</DropdownMenuItemLabel>
```

Comment:
![image](https://github.com/WordPress/gutenberg/assets/54422211/95db579d-186b-49b4-beaf-ab2da73f6ad5)

Like the shadow presets, "Delete" might be good. Also, how about moving the rename function here.

Reply:
updated here: [`0b6c490` (#62328)](https://github.com/WordPress/gutenberg/pull/62328/commits/0b6c490712fd1ded4e9a641a869051b5e2c1e42d)